vmm/src/migration.rs

```diff
@@ -49,10 +49,12 @@ pub fn recv_vm_config(source_url: &str) -> std::result::Result<VmConfig, Migrata
     vm_config_path.push(SNAPSHOT_CONFIG_FILE);
 
     // Try opening the snapshot file
-    let vm_config_file =
-        File::open(vm_config_path).map_err(|e| MigratableError::MigrateSend(e.into()))?;
-    let vm_config_reader = BufReader::new(vm_config_file);
-    serde_json::from_reader(vm_config_reader).map_err(|e| MigratableError::MigrateReceive(e.into()))
+    let mut vm_config_file =
+        File::open(vm_config_path).map_err(|e| MigratableError::MigrateReceive(e.into()))?;
+    let mut bytes = Vec::new();
+    vm_config_file.read_to_end(&mut bytes).unwrap();
```

Comment:
It's better to propagate the error and let the caller decide what to do, say `vm_config_file.read_to_end(&mut bytes).map_err(|e| MigratableError::MigrateReceive(e.into()))?;`

Reply:
Updated, thx.